src/v2/components/design/modal/Modal.tsx

```diff
@@ -1,9 +1,54 @@
+import { useCallback, useEffect } from 'react';
+import { createPortal } from 'react-dom';
 import styled from 'styled-components';
 
-export type ModalProps = {
+export type modalProps = {
 	children?: JSX.Element;
+	isShowing?: boolean;
+	closeModal: () => void;
 };
-export const Modal = ({ children }: ModalProps): JSX.Element => {
-	return <StyledContainer>{children}</StyledContainer>;
+
+const Modal = ({ children, isShowing, closeModal }: modalProps) => {
```

Comment:
isShowing prop seems redundant here. We can manage the Modal visibility from parent component itself.
```suggestion
const Modal = ({ children, onCloseModal }: modalProps) => {
```